### PR TITLE
CATTY-109 Convert FormulaTest from ObjC to Swift

### PR DESCRIPTION
**Pull request by aeluna**
Convert FormulaTest from ObjectiveC to Swift.

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#ios-general* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by m-herold** [comment]
Please resolve your merge conflicts. Thanks!

**Comment by m-herold** [comment]
retest this please

**Comment by m-herold** [comment]
Jenkins: retest this please

**Comment by m-herold** [comment]
Jenkins: retest this please

File: src/CattyTests/Formula/FormulaTest.swift [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        tokenNumber = InternToken(type: TOKEN_TYPE_NUMBER, andValue: "1.0")
+        let secondToken = InternToken(type: TOKEN_TYPE_OPERATOR, andValue: MinusOperator.tag)!
+        let secondNumber = InternToken(type: TOKEN_TYPE_NUMBER, andValue: "1.0")!
+        internTokenList.add(token)
+        internTokenList.add(tokenNumber)
+        internTokenList.add(secondToken)
+        internTokenList.add(secondNumber)
+
+        internParser = InternFormulaParser(tokens: (internTokenList as! [InternToken]), andFormulaManager: formulaManager)
+        parseTree = internParser!.parseFormula(for: nil)
```
Comment (GatCode) [inline src/CattyTests/Formula/FormulaTest.swift]: Please try to reduce the overall amount of force unwraps in this file, thank you!

Comment (aeluna) [reply src/CattyTests/Formula/FormulaTest.swift]: Thank you for the tip!
I think I fixed all the force unwraps.

File: src/CattyTests/Formula/FormulaTest.swift [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        let screenSize = Util.screenSize(true)
+        formulaManager = FormulaManager(sceneSize: screenSize)
+        interpreter = formulaManager
+    }
+
+    func testIsSingularNumberFormula() {
+        //var formula = Formula(integer: 1)
+        if let formula = Formula(integer: 1) {
+            XCTAssertTrue(formula.isSingularNumber())
+            print("yes")
```
Comment (m-herold) [inline src/CattyTests/Formula/FormulaTest.swift]: Please remove print message and comment

Comment (aeluna) [reply src/CattyTests/Formula/FormulaTest.swift]: Done. I totally overlooked it. Thank you for the hint!

**Pull request merged**